sql/core.go

```diff
@@ -314,6 +314,18 @@ type PrimaryKeySchemaTarget interface {
 	WithPrimaryKeySchema(schema PrimaryKeySchema) (Node, error)
 }
 
+// TableFunction is a node that is generated by a function
+type TableFunction interface {
+	Node
+	Expressioner
+	Databaser
+
+	// NewInstance returns a new instance of the table function
+	NewInstance(Database, []Expression) (Node, error)
```

Comment:
In general I like to name the params in interface method declarations, because it gives better hints for code generation in IDEs.

Reply:
nice tip; I'll follow that pattern.